### PR TITLE
[8.x] Testing: Access component properties from the return value of $this->component()

### PR DESCRIPTION
**Pull request by yob-yob**
The benefits for this pull-request would be that it's possible to access component attributes when using 
`$this->component()` in your test

Example:

```
$table = $this->component(Table::class, [ 'data' => ['foo','bar','baz','quz'] ]);

$string = $table->concatenated_data;

$this->assertEquals( 'foobarbazquz', $string );
```

It's not going to break any existing features because It's still possible to use the assertions from `TestView::class` since `TestComponent::class` it extends to that class...

I know that it's possible to use `$this->app->make($componentClass, $data);` to instantiate and access properties from a component but it does not have the assertions that `$this->component()` have.... 
if you want to access the component properties and use the assertions you need to do both...
also `$this->app->make($componentClass, $data);` is not documented.

I was looking to the test for the `TestView::class` but I didn't found any so I just added it into `InteractsWithViewsTest`

It's my first contribution, and I'm open to constructive criticism.

### CODE REVIEW

**Comment by driesvints** [comment]
Hi, I merged https://github.com/laravel/framework/pull/38398 into 8.x to get the test suite passing again. I've put your PR in draft. Please rebase your PR against 8.x to make the tests of this PR pass. After you've done that and tests pass, mark this PR as ready for review. Thanks.

**Comment by yob-yob** [comment]
So I need to support lower versions of php8.0 😅 My Bad... will update the code

**Comment by yob-yob** [comment]
@driesvints  just a question... before I touch anything else... the errors are showing that tests are failing... but on test that this pull-request has not even touched... is this normal? and should I modify and fix those test files even thou It's not concerned with my pull-request in the first place?

I also checked...there's a commit from 2 days ago with a comment of "Apply Fixed from Style-CI"

or is there something wrong here that I'm not catching-up?

**Comment by yob-yob** [comment]
@driesvints  sorry about that... I just needed to fetch from upstream... I forked a commit that has a failing test... and forgot to fetch from upstream before making the request... 

I read about the Contribution Guide.. [and it says that I shouldn't worry about StyleCI Errors since it would automatically fix styling](https://laravel.com/docs/8.x/contributions#styleci)

**Comment by GrahamCampbell** [comment]
> I read about the Contribution Guide.. and it says that I shouldn't worry about StyleCI Errors since it would automatically fix styling

Yes, it will. When the PR is merged, the fixes will be applied.

File: src/Illuminate/Foundation/Testing/Concerns/InteractsWithViews.php [hunk 1]
```diff
@@ -33,13 +34,13 @@ protected function blade(string $template, array $data = [])
     {
         $tempDirectory = sys_get_temp_dir();
 
-        if (! in_array($tempDirectory, ViewFacade::getFinder()->getPaths())) {
+        if (!in_array($tempDirectory, ViewFacade::getFinder()->getPaths())) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Testing/Concerns/InteractsWithViews.php:37]: please revert unrelated changes

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Foundation/Testing/Concerns/InteractsWithViews.php (hunk 1 above)
Comment (yob-yob) [reply src/Illuminate/Foundation/Testing/Concerns/InteractsWithViews.php:37]: ahhh... my editor fixed the styling when I save...
